--- 1.22.6/src/github.com/juju/juju/testcharms/charm.go.orig	2015-06-24 12:02:02.746416146 +0200
+++ 1.22.6/src/github.com/juju/juju/testcharms/charm.go	2015-06-24 12:03:49.810418650 +0200
@@ -10,4 +10,6 @@
 )
 
 // Repo provides access to the test charm repository.
-var Repo = testing.NewRepo("charm-repo", "quantal")
+// XXX fake-juju: avoid crashing because the charm-repo dir is not there
+//var Repo = testing.NewRepo("charm-repo", "quantal")
+var Repo = &testing.Repo{}

--- 1.22.6/src/github.com/juju/juju/provider/dummy/environs.go.orig	2015-07-06 15:01:14.200568258 +0200
+++ 1.22.6/src/github.com/juju/juju/provider/dummy/environs.go	2015-07-06 15:18:32.648549661 +0200
@@ -628,9 +628,9 @@
 
 // PrecheckInstance is specified in the state.Prechecker interface.
 func (*environ) PrecheckInstance(series string, cons constraints.Value, placement string) error {
-	if placement != "" && placement != "valid" {
-		return fmt.Errorf("%s placement is invalid", placement)
-	}
+//	if placement != "" && placement != "valid" {
+//		return fmt.Errorf("%s placement is invalid", placement)
+//	}
 	return nil
 }
 
--- 1.22.6/src/github.com/juju/juju/testing/cert.go	2016-03-18 09:25:34 +0000
+++ 1.22.6/src/github.com/juju/juju/testing/cert.go	2016-03-18 09:26:04 +0000
@@ -49,7 +49,7 @@
 }
 
 func mustNewCA() (string, string) {
-	cert.KeyBits = 512
+	cert.KeyBits = 1024
 	caCert, caKey, err := cert.NewCA("juju testing", time.Now().AddDate(10, 0, 0))
 	if err != nil {
 		panic(err)
@@ -58,7 +58,7 @@
 }
 
 func mustNewServer() (string, string) {
-	cert.KeyBits = 512
+	cert.KeyBits = 1024
 	var hostnames []string
 	srvCert, srvKey, err := cert.NewServer(CACert, CAKey, time.Now().AddDate(10, 0, 0), hostnames)
 	if err != nil {

